### PR TITLE
Restructure CLI section

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

Restructure CLI section in a similar style to https://pip.pypa.io/en/stable/cli/

- Old CLI Section to be removed.
- Created 'Commands' section under 'Advanced reference'.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/advanced_reference/commands/autonomy_run.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## Run
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_run.md:1]: This does not belong here, we're not extending the `run` command in any way

File: docs/advanced_reference/commands/autonomy_scaffold.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+## Scaffold
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_scaffold.md:1]: We should only have documentation from FSM scaffolding, the logic for scaffolding other components belong on open-aea

File: docs/advanced_reference/commands/autonomy_analyse.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+Options:
+  -b, --block-type [local|consensus|total|all]
+  -d, --period INTEGER
+  -o, --output FILE
+  --help                          Show this message and exit.
+```
+
+Aggregating results from deployments.
+
+To use this tool you'll need benchmark data generated from agent runtime. To generate benchmark data run
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_analyse.md:123]: Create a ticket for benchmarking guide in the ice box

File: docs/advanced_reference/commands/autonomy_deploy.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
+
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_deploy.md:1]: Add a small section on what this command group is about at the top

File: docs/advanced_reference/commands/autonomy_deploy.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+## Usage
+
+```bash
+autonomy deploy [OPTIONS] COMMAND [ARGS]
+```
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_deploy.md:6]: No need for a usage section on a parent command unless there are some top level flags

File: docs/advanced_reference/commands/autonomy_deploy.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+## Description
+
+Build or run agent service deployments.
+
+This meta-command aggregates a number of functionalities for building service deployments, run locally stored service deployments, and run service deployments defined in the on-chain protocol. See the appropriate subcommands for more information.
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_deploy.md:12]: Should be at the top

File: docs/advanced_reference/commands/autonomy_deploy.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+
+## Options
+`--help`
+
+:   Show the help message and exit.
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_deploy.md:18]: No need for the section if only flag is the `--help` flag

File: docs/advanced_reference/commands/autonomy_deploy.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+
+
+
+
+## Subcommand `autonomy deploy build`
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_deploy.md:23]: No need to specify the word `subcommand`

File: docs/advanced_reference/commands/autonomy_replay.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+`--help`
+:   Show the help message and exit.
+
+
+## Examples
+
+The example below shows how to replay agent runs from Tendermint dumps.
+
+
+1.  Copy and paste the following Makefile into your workspace folder:
```
Comment (angrybayblade) [inline docs/advanced_reference/commands/autonomy_replay.md:62]: This is wrong, this example just create a dev setup and uses it for replays later. Please redirect the user to the development setup guide and then explain how to use the replays on the dev setup.

Comment (jmoreira-valory) [reply docs/advanced_reference/commands/autonomy_replay.md]: Will add a ticket to address in another PR

File: docs/advanced_reference/commands/autonomy_deploy.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+`--password PASSWORD`
+:   Set password for key encryption/decryption.
+
+`--help`
+:   Show the help message and exit.
+
+
+### Examples
+```bash
+autonomy deploy build keys.json --aev
```
Comment (dvilelaf) [inline docs/advanced_reference/commands/autonomy_deploy.md:80]: --aev is the default now, it is not required to specify it AFAIK. Also keys.json.

**Review by DavidMinarsch — APPROVED** [review]

File: docs/advanced_reference/commands/autonomy_analyse.md [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+
+
+
+## `autonomy analyse docstrings `
+
+Analyse {{fsm_app}} skill docstring definitions.
+
+This command verifies that the [`AbciApp` class](../../key_concepts/abci_app_class.md) docstring is follows a standard format.
+
+??? example
```
Comment (Karrenbelt) [inline docs/advanced_reference/commands/autonomy_analyse.md:13]: ???

Comment (jmoreira-valory) [reply docs/advanced_reference/commands/autonomy_analyse.md:13]: mkdocs-material format airtfact.

**Pull request merged**